exchanges/stream/stream_match.go

```diff
@@ -5,76 +5,69 @@ import (
 	"sync"
 )
 
+var (
+	errSignatureCollision = errors.New("signature collision")
+	errInvalidBufferSize  = errors.New("buffer size must be positive")
+)
+
 // NewMatch returns a new Match
 func NewMatch() *Match {
-	return &Match{
-		m: make(map[interface{}]chan []byte),
-	}
+	return &Match{m: make(map[any]incoming)}
 }
 
 // Match is a distributed subtype that handles the matching of requests and
 // responses in a timely manner, reducing the need to differentiate between
 // connections. Stream systems fan in all incoming payloads to one routine for
 // processing.
 type Match struct {
-	m  map[interface{}]chan []byte
+	m  map[any]incoming
```

Comment:
```suggestion
	m  map[any]*incoming
```

Please look over your test `TestMatch` again. I've snipped irrelevant bits

```go
	ch, err := match.Set("hello", 2)
	assert.True(t, match.IncomingWithData("hello", []byte("hello")))
	assert.Equal(t, "hello", string(<-ch))
	assert.True(t, match.IncomingWithData("hello", load), "Should match with matching message and signature")
	assert.True(t, match.IncomingWithData("hello", load), "Should match with matching message and signature")

	assert.Len(t, ch, 2, "Channel should have 2 items")
```

Note that by not having a pointer in your map, `expected` never sets its `expected--`, so the _second_ time it goes in, `expected` is still `2` despite having a match already.

The scenario you've made in the test should not be able to occur because the signature should already be deleted by the third time you call `hello`